serializer.hh

```diff
@@ -303,6 +303,61 @@ struct normalize<bytes_ostream> {
 template <typename T, typename U>
 struct is_equivalent : std::is_same<typename normalize<std::remove_const_t<std::remove_reference_t<T>>>::type, typename normalize<std::remove_const_t <std::remove_reference_t<U>>>::type> {
 };
+
+// gc_clock duration values were serialized as 32-bit prior to 3.1, and
+// are serialized as 64-bit in 3.1.0. To allow communication with both
+// 3.0.* (and prior) and 3.1.0, we define the following serialization format:
+//
+// extern bool gc_clock_using_3_1_0_serialization;
+//
+// if (gc_clock_using_3_1_0_serialization) {
+//     use 64-bit format
+// } else {
+//     if (value fits in 32-bits) {
+//         use 32-bit format
+//     } else {
+//         write 0x81828384 magic value
+//         write 64-bit value
+//     }
+// }
+//
+// This relies on gc_clock values of 0x81828384 not being legal as
+// they are deeply negative.
+
+inline bool gc_clock_using_3_1_0_serialization = false;
+
+constexpr int32_t gc_clock_magic_value = int32_t(0x81828384);
+
+template <typename Output>
+void
+serialize_gc_clock_duration_value(Output& out, int64_t v) {
+    if (gc_clock_using_3_1_0_serialization) {
+        serializer<int64_t>::write(out, v);
+    } else {
+        if (int32_t(v) == v) {
+            serializer<int32_t>::write(out, int32_t(v));
+        } else {
+            serializer<int32_t>::write(out, gc_clock_magic_value);
+            serializer<int64_t>::write(out, v);
```

Comment:
This assumes that there will be no mixed clusters by the time a legit TTL can overflow the 32 bit value.

Reply:
As @bhalevy TTLs can't really overflow. 31-bit == 68 years. Only time_points can overflow.

Reply:
So we can just serialize as 32-bit, unless 3.1.0. No need for 12-byte serialization.